tests/test_population.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+from scip.api import population
+
+# test data
+from sample_data import PCH1, PCH2, PPKO, PPKE, check_populations
+
+
+# check listing of all populations
+def test_population_listing(db_populated_session):
+    response = population(db_populated_session)
+    assert check_populations([PCH1, PCH2, PPKO, PPKE], response)
+
+
+# check listing by species
+@pytest.mark.parametrize(
+    "species,expected", [["Chum", [PCH1, PCH2]], ["Pink", [PPKO, PPKE]], ["Coho", []]]
+)
+def test_population_by_species(db_populated_session, species, expected):
+    response = population(db_populated_session, common_name=species)
+    assert check_populations(expected, response)
+
+
+# check listing by subgroup
+@pytest.mark.parametrize(
+    "species,subgroup,expected",
+    [
+        ["Chum", None, [PCH1, PCH2]],
+        ["Chum", "Banana", []],
+        ["Chum", "Odd", []],
+        ["Pink", "Odd", [PPKO]],
+        ["Pink", "Even", [PPKE]],
+        [None, "Even", [PPKE]],
+    ],
+)
+def test_population_by_subgroup(db_populated_session, species, subgroup, expected):
+    response = population(db_populated_session, common_name=species, subgroup=subgroup)
+    assert check_populations(expected, response)
+
+
+@pytest.mark.parametrize(
+    "x,y,species,expected",
+    [
+        [0, 0, None, [PPKE, PCH1]],
+        [0, 0, "Chum", [PCH1]],
+        [15, 15, None, [PPKE, PCH2]],
+        [15, 15, "Pink", [PPKE]],
+        [100, 100, None, []],
+        [30, 30, "Pink", [PPKO]],
+        [30, 30, None, [PPKO]],
+        [30, 30, "Chum", []],
+    ],
+)
+def test_population_by_overlap_point(db_populated_session, x, y, species, expected):
+    wkt = "POINT({} {})".format(x, y)
+    response = population(db_populated_session, common_name=species, overlap=wkt)
+    assert check_populations(expected, response)
+
+
+@pytest.mark.parametrize(
+    "species,expected",
+    [
+        [None, [PCH2, PPKO, PPKE]],
+        ["Pink", [PPKO, PPKE]],
+        ["Chum", [PCH2]],
+        ["Banana", []],
+    ],
+)
+def test_population_by_overlap_polygon(db_populated_session, species, expected):
+    wkt = "POLYGON((15 15, 15 20, 20 20, 20 15, 15 15))"
```

Comment:
Can you explain this test?

Reply:
This test is similar to the previous test, `test_population_by_overlap_point`, which finds salmon populations that contain a point, except that instead of looking for populations present at a point, it looks for populations that overlap any part of an arbitrary polygon.

I wanted to include this test because it more closely matches how we anticipate users actually accessing the API than the previous test involving points. Our anticipated users are people tasked with managing salmon streams across some sort of civil area, like a tribal area or a municipality or something. We plan people will select a named polygon from a menu, or draw it on the map, and want to see all salmon populations that overlap any part of their polygon.

In practice, PostGIS treats points and polygons identically when calculating overlap, but I wanted to include this "extra" test case with a hard-coded polygon anyway, just so we had something that matched how we were actually going to use this API.

Reply:
Am I missing something? The one and only line in the test function that I can see is

```
wkt = "POLYGON((15 15, 15 20, 20 20, 20 15, 15 15))"
```

It doesn't appear to test anything.

Reply:
Oh! I apparently forgot to write the rest of it. Oops. I'll get that up!